File: app/src/main/java/com/moducode/gw2serveralarm/schedulers/BaseSchedulerProvider.java
package com.moducode.gw2serveralarm.schedulers;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by Jay on 2017-08-20.
 */

public class BaseSchedulerProvider implements SchedulerProvider {

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }

    @Override
    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler compute() {
        return Schedulers.computation();
    }
}

File: app/src/main/java/com/moducode/gw2serveralarm/schedulers/BaseSchedulerProvider.kt
package com.moducode.gw2serveralarm.schedulers

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by Jay on 2017-08-20.
 */

class BaseSchedulerProvider : SchedulerProvider {

    override fun io(): Scheduler = Schedulers.io()

    override fun ui(): Scheduler = AndroidSchedulers.mainThread()

    override fun compute(): Scheduler = Schedulers.computation()

}
